api/proto/teleport/accesslist/v1/accesslist.proto

```diff
@@ -71,6 +71,12 @@ message AccessListSpec {
 
   // owner_grants describes the access granted by owners to this Access List.
   AccessListGrants owner_grants = 11;
+
+  // owner_of describes Access Lists that this Access List is an explicit owner of.
+  repeated string owner_of = 12;
```

Comment:
Do we need to keep track of this information for every access list?

Reply:
My concern here is that AL being imported by sync utils will have to preserve these fields when they grant access to Teleport access lists. 
If we don't need to specify these here and can compute them when needed, it would be much better.

Reply:
yes, these need to be tracked for every list. I don't love either option... computing when needed is what was initially implemented, but it was much more expensive and required passing around all existing access lists for checks

Reply:
Can we store it under different keys if the goal is to keep it internal? 

Or move it to the status field

Reply:
Status aren't meant to be edited by users and you can always keep the state there. We can probably keep some logic to keep it when reconcilers run

Reply:
Is the `status` field persisted?

Reply:
Yes, we persist status fields.

Reply:
No objections from me to moving these fields to `status` as users aren't supposed to set them.

Reply:
Moved them into `status`, it wasn't persisted previously so I named the json/yaml field as `status` and set the member/nested list count fields within it to `-`. LMK if it looks good